src/ssort/_files.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import os
+import pathlib
+from typing import Iterable
+
+import pathspec
+
+from ssort._utils import memoize
+
+_EMPTY_PATH_SPEC = pathspec.PathSpec([])
+
+
+@memoize
```

Comment:
This, along with the conversation about API stability, reminds me why I tried to implement that `memoize_weak` decorator:  what happens if someone wants to call `ssort.ssort` multiple times from a long running process.  `memoize_weak` is not the answer in this case because you will get new `Path` instances for each traversal, but there should be some way to make sure that `.gitignore` files are not cached between runs.

Anyway, it's completely irrelevant here because the `find_python_files` function is only called directly from main.  `@memoize` is the best solution.

Reply:
I see. This could be solved by using a more object-oriented approached where `.gitignore` files are cached within some object instance instead of globally. However, such a change seems out of scope for this PR. Definitely something to keep in mind though.